nvm.sh

```diff
@@ -2413,11 +2413,19 @@ nvm() {
       fi
 
       for tool in git grep awk sed cut basename rm mkdir xargs; do
+        local NVM_TOOL_VERSION
         if nvm_has "${tool}"; then
-          nvm_err "${tool}: $(nvm_command_info ${tool}), $(command ${tool} --version | command head -n 1)"
+          NVM_TOOL_VERSION="$(command ${tool} --version 2>&1 | command head -n 1)"
+          if echo "${NVM_TOOL_VERSION}" | command grep -q 'illegal option'; then
```

Comment:
Can you elaborate on when "illegal option" shows up?

Reply:
@ljharb it's on macOS 😞

Reply:
I do see that `sed --version` fails on my Mac; in that case, it simply shouldn't be in this `for` loop. Only simple, straightforward cases should be in this loop.

Reply:
so we should determinate the commands to be tested outside the loop?

Reply:
No, I'm saying that if `sed --version` doesn't work everywhere, it should never be in the loop - it should have a hardcoded section to handle it.

Reply:
What about `if command --version 2>&1 > /dev/null; then print-command-version; fi` ?

Reply:
I'd prefer to keep `nvm debug` as simple as possible, even if that makes it more verbose.

Reply:
So we simply remove those commands without `--version` support?

Reply:
Or, have a separate, manual section for `sed`, and another repeated one for each command that's not universal.

Reply:
Just to make sure, so they'll looks like the part as curl & wget, am I correct? I can fix it here if I'm right. Thanks.

Reply:
Yes, basically that :-)

Reply:
@ljharb for `sed`, `cut`, `basename`, `rm`, `mkdir` & `xargs`, would you like to have their version info on non-macOS?

Actually `wget -V` & `curl -V` may not work everywhere, for example, on busybox based environment, so I still barely know how should I implement this one.

Looks like trying to catch if the command supports `-V` is not what you want, so the rule/pattern for each may be static, and even based on OS?

Reply:
Yes, exactly

Reply:
Like this: On macOS & FreeBSD, do not try `-V` parameter?